src/plugins/visualizations/public/content_management/visualization_client.ts

```diff
@@ -66,6 +66,14 @@ const deleteVisualization = async (id: string) => {
 };
 
 const search = async (query: SearchQuery = {}, options?: VisualizationSearchQuery) => {
```

Comment:
Does the return type of this method account for that it could return `maps` and `lens` objects?

Reply:
Not specifically the types of those, but all adhere to a generic interface defined https://github.com/elastic/kibana/pull/165485/files#diff-72eb8e94b67c0c56b5744ac732c6164a66b68e6e4a6fb1857da4d361acb6b149R52

Having the exact types returned by this method, from my understanding, requires either a type cast or a full refactor of the architecture, as types are registered dynamically to Visualizations and there's no static way to detect the correct type.